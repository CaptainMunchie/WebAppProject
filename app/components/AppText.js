import React from 'react';

import { StyleSheet, Text, Platform } from 'react-native';

function AppText({style, children}) {
    return (
        <Text style={[styles.text, style]}>{children}</Text>
    );
}

const styles = StyleSheet.create({
    text:{
        fontSize: 20,
        fontFamily: Platform.OS ==='android' ? "monospace" : "Cochin",
        paddingLeft:10,
        justifyContent:"center",
        alignItems:"center",
        
        
    },  
})

export default AppText;