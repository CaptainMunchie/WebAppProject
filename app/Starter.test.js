import React from 'react';
import renderer from 'react-test-renderer';
import AppText from './components/AppText';

test("trivially true", () => {
    expect(1).toBe(1);
});

test("AppTest will render", () => {
    const tree = renderer.create(<AppText/>).toJSON();
    console.log(tree.props.style);
    expect(tree.props.style[0].fontSize).toBe(20);
    expect(tree.props.style[0].justifyContent).toBe("center");
    expect(tree.props.style[0].alignItems).toBe("center");
    expect(tree.props.style[0].fontFamily).toBe("Cochin");
});

test("Apptext contains text", () => {
    const tree = renderer.create(<AppText>Text data can be input</AppText>).toJSON();
    expect(tree.props.style[0].fontSize).toBe(20);
    expect(tree.props.style[0].fontFamily).toBe("Cochin");
    expect(tree.children.includes("Text data can be input"));
});

test("SnapShot expect for no changes", () => {
    const tree = renderer.create(<AppText/>).toJSON();
    expect(tree).toMatchSnapshot();
});



