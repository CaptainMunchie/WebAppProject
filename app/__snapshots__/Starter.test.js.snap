// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapShot expect for no changes 1`] = `
<Text
  style={
    Array [
      Object {
        "alignItems": "center",
        "fontFamily": "Cochin",
        "fontSize": 20,
        "justifyContent": "center",
        "paddingLeft": 10,
      },
      undefined,
    ]
  }
/>
`;
